cardano-api/src/Cardano/Api.hs

```diff
@@ -263,6 +266,18 @@ shelleyVerificationKeyAddress vkey _nw =
         Shelley.Addr (Shelley.KeyHashObj (Shelley.hashKey vk))
                      Shelley.StakeRefNull
 
+-- | Builds a shelley base address. This type of address has staking rights and
+-- therefore requires a payment verification key and a staking verification key.
+shelleyVerificationKeyBaseAddress :: ShelleyVerificationKey -> ShelleyVerificationKeyStaking -> ShelleyAddress
```

Comment:
I think I'd stick to one name, e.g. `shelleyVerificationKeyAddress` but stick a `Maybe` on the `ShelleyVerificationKeyStaking`.